crates/client/rpc/src/lib.rs

```diff
@@ -652,4 +652,113 @@ where
+                    if txn.hash != transaction_hash.into() {
+                        continue;
+                    }
+
+                    return Ok(Transaction::try_from(txn.clone()).map_err(|e| {
+                        error!("{:?}", e);
+                        StarknetRpcApiError::InternalServerError
+                    })?);
+                }
+
+                Err(StarknetRpcApiError::TxnHashNotFound.into())
+            }
+            BlockTransactions::Hashes(_hashes) => {
+                // TODO: Here what can I do? I don't have all the info,
+                // so what's the expected result, TxnNotFount? Error?
+                Err(StarknetRpcApiError::TxnHashNotFound.into())
+            }
+        }
+    }
+
+    /// Returns the receipt of a transaction by transaction hash.
+    ///
+    /// # Arguments
+    ///
+    /// * `transaction_hash` - Hex string of the transaction hash.
+    fn get_transaction_receipt(&self, transaction_hash: FieldElement) -> RpcResult<MaybePendingTransactionReceipt> {
+        let block_hash_from_db = self
+            .backend
+            .mapping()
+            .block_hash_from_transaction_hash(H256::from(transaction_hash.to_bytes_be()))
+            .map_err(|e| {
+                error!("Failed to get transaction's substrate block hash from mapping_db: {e}");
+                StarknetRpcApiError::TxnHashNotFound
+            })?;
+
+        let substrate_block_hash = match block_hash_from_db {
+            Some(block_hash) => block_hash,
+            None => {
+                // TODO: verify if the tx is in the tx pool (self.pool.ready()).
+                //       Question: If the transaction is in the mempool, it's not actually
+                //       Pending status. Because we don't have any receipt at this point.
+                //       What is the expected behavior? TxnHashNotFound?
+                return Err(StarknetRpcApiError::TxnHashNotFound.into());
+            }
+        };
+
+        let block = self
+            .overrides
+            .for_block_hash(self.client.as_ref(), substrate_block_hash)
+            .current_block(substrate_block_hash)
+            .unwrap_or_default();
+
+        match block.transactions() {
+            BlockTransactions::Full(transactions) => {
+                for (_, receipt) in transactions {
+                    if receipt.transaction_hash != transaction_hash.into() {
+                        continue;
+                    }
+
+                    return Ok(receipt.clone().into_maybe_pending_transaction_receipt(TransactionStatus::AcceptedOnL2));
+                }
+
+                Err(StarknetRpcApiError::TxnHashNotFound.into())
+            }
+            BlockTransactions::Hashes(_hashes) => {
+                // TODO: Here what can I do? I don't have all the info,
+                // so what's the expected result, TxnNotFount? Error?
+                Err(StarknetRpcApiError::TxnHashNotFound.into())
+            }
+        }
+    }
```

Comment:
Can probably be factored with the previous one

Reply:
Refactored using find as you suggested.